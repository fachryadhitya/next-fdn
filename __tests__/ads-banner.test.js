import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import AdsBanner from '../components/ads-banner';

describe('Ads Banner Component', () => {
  it('Render successfully', () => {
    const { container } = render(<AdsBanner width={700} height={700} />);
    expect(container).toMatchSnapshot();
  });

  it('Render component with specific width', () => {
    const { getByTestId, getByText } = render(<AdsBanner width={700} height={700} />);
    expect(getByTestId('ads-banner')).toHaveStyle('width: 700px');
    expect(getByText('700 x 700')).toBeInTheDocument();

  });
});
