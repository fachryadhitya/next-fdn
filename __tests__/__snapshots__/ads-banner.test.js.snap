// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ads Banner Component Render successfully 1`] = `
<div>
  <div
    class="bg-gray-200 text-center flex justify-center items-center my-5 max-w-full mx-auto"
    data-testid="ads-banner"
    style="width: 700px; height: 700px;"
  >
    <p>
      700
       x 
      700
    </p>
  </div>
</div>
`;
